packages/medusa/src/services/totals.js

```diff
@@ -0,0 +1,28 @@
+import _ from "lodash"
+import { BaseService } from "medusa-interfaces"
+
+/**
+ * A service that calculates total and subtotals for orders, carts etc..
+ * @implements BaseService
+ */
+class TotalsService extends BaseService {
+  cartSubTotal(cart) {
```

Comment:
Alright.. So we assume, that developers know, that getSubtotal is related to carts? What if we at some point get something else, to which a getSubTotal would be relevant?

Reply:
The totals service only works for `carts` and `orders`. It will take the object it is being passed and look at the `items` field and sum the line items. If at some point we find that it would make sense to calculate the subtotal for some other object we could still use the `getSubtotal` method as an abstraction and then call the correct private methods depending on what object is being used as input.